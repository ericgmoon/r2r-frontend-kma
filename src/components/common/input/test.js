import React from "react";
import renderer from "react-test-renderer";
import Input from "./index";

describe("renders correctly", () => {
  test("without props", () => {
    const node = renderer.create(<Input />).toJSON();
    expect(node).toMatchSnapshot();
  });

  test("with custom styles", () => {
    const node = renderer
      .create(<Input style={{ paddingTop: 32, paddingButton: 32 }} />)
      .toJSON();
    expect(node).toMatchSnapshot();
  });
});
