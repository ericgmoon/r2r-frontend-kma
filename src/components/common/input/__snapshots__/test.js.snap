// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with custom styles 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 2,
        "borderColor": "#000",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 2,
        "flexDirection": "row",
        "paddingButton": 32,
        "paddingTop": 32,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      placeholderTextColor="#86939e"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "color": "#242424",
          "flex": 1,
          "fontSize": 32,
          "fontWeight": "bold",
          "minHeight": 40,
          "padding": 12,
        }
      }
      testID="RNE__Input__text-input"
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;

exports[`renders correctly without props 1`] = `
<View
  style={
    Object {
      "paddingHorizontal": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 2,
        "borderColor": "#000",
        "borderRadius": 10,
        "borderStyle": "solid",
        "borderWidth": 2,
        "flexDirection": "row",
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": "100%",
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      editable={true}
      placeholderTextColor="#86939e"
      rejectResponderTermination={true}
      style={
        Object {
          "alignSelf": "center",
          "color": "#242424",
          "flex": 1,
          "fontSize": 32,
          "fontWeight": "bold",
          "minHeight": 40,
          "padding": 12,
        }
      }
      testID="RNE__Input__text-input"
      underlineColorAndroid="transparent"
    />
  </View>
  <Text
    style={
      Object {
        "color": "#ff190c",
        "fontSize": 12,
        "margin": 5,
      }
    }
  />
</View>
`;
